### DISCUSSION TITLE
252优化

### DISCUSSION THREAD

**Discussion by Shawnsdaddy** [Konano/arknights-mower#515] [General]
1. 自动根据当前基建干员心情调整睡觉策略（比如说大家心情都过低则尽可能全安排进宿舍，否者只利用4个VIP位）
      这个将会是我的主要开发方向，因为我觉得252 得 24/7  确保VIP位置的单回被高效组吃到
3. 额外读秒回满心情时间，及时释放宿舍位
4. 如果说当前宿舍空位过多（可能由于分组原因导致），则可以考虑提前执行上班任务？来安排更多的人进宿舍？？
5. 252的急救模式应该改成休息1.5小时（0.5小时回不了多少）
6. 如果同组干员的回复时间差距过大，则可以提前释放同组已经休息好的让剩下的继续休息

**Comment by ghzh21th** [comment]
线索葛朗台模式：获赠线索先读取，数量少的线索不收，置空等好友赠送。开趴前半小时再单独收取缺失线索

**Reply by ghzh21th** [reply]
252信用经常不够买

**Reply by Shawnsdaddy** [reply]
不收，好友还是知道你有的。可以开小号试试

**Reply by ghzh21th** [reply]
b服小号实测结果是，不收看不到有。收了之后才会亮，挂不挂上不影响

**Reply by Luoxcat** [reply]
不如多几个小号自动登录送线索

**Comment by 383175186** [comment]
优化回满逻辑。

其实更像是一个自动低优先的设计。在此基础上可以增加进宿舍读秒后，组内心情高于23的直接丢出宿舍。空出宿舍位子。

比如同组高优统计回满时间，有人最早回满时间比第二个早一两个小时，则忽略最早一人，换班时间以第二人为准。这样可以减少直接设置回满人数，更多的交给mower去进行换班调整，而不会出现多个回满卡位导致无法换班的情况。
不过要增加对令夕的单独处理，感知烟火的情况下，回到12就丢出宿舍，避免恢复超额。

使用场景1，252自动化，森蚺中枢，有骑士套的减免导致心情消耗低于格雷伊，但是设置低优先的情况下，又不足以在格雷伊回满时回满自身。所以通常使用条件是设置整组高优格雷伊回满。在此基础上如果有多个回满比如伊内丝稀音，刚好三者休息时间头接尾，则会导致长时间VIP位低于4人，导致其他需求4高优的套组无法换班。

使用场景2，252烟火不带琴柳，正常设置是只有中枢三人高优，如果设置了第四人，则会出现高优四人 三人低心情，一人高心情，这时候会出现高心情回满直接换班，导致整组心情开始崩溃。

使用场景3，同样适用于一些套组没写好的，或者更换作业时，新增人员心情未对齐的情况下，换班混乱的状态。

**Comment by fuyn101** [comment]
给常驻宿舍的buff人（某些杜林）绑组解放空间腾位置
给加工站（年 芳烃 白铁）等搞特殊优化
<img width="815" alt="e06b744ca21837d812847bd1da133f06" src="https://github.com/ArkMowers/arknights-mower/assets/151902446/08058aac-2f6b-4bbf-abb8-ac84889b1b4e">
 比如 我在用完芳烃 把他塞到红框这个位置
然后mower换班的时候 随便找个free把他顶掉了

**Comment by ZhaoZuohong** [comment]
252的主要问题是要不要考虑 11贸（3宿舍）。11贸面向有但书的低配，用的人实际上非常少（高配压贸易可以用31上裁缝），又可以靠增加最大组人数解决，所以我觉得可以不考虑 3 宿舍的布局。以下内容基于 4 宿舍。

> 得 24/7 确保VIP位置的单回被高效组吃到

这是不可能实现的，4 个单回位的恢复能力大于高效组的需求。

> 额外读秒回满心情时间，及时释放宿舍位

1. 释放宿舍位后重启 mower，纠错是个问题
2. 释放宿舍位后得立即安排休息才能用上释放了的宿舍位。一般来说，单回之间的差距不值得这样操作，群回在安排休息之前看有没有回满的就行。

> 当前宿舍空位过多（可能由于分组原因导致），则可以考虑提前执行上班任务？来安排更多的人进宿舍？？

想不到相关的场景

> 252的急救模式应该改成休息1.5小时（0.5小时回不了多少）

想改就改吧

> 如果同组干员的回复时间差距过大，则可以提前释放同组已经休息好的让剩下的继续休息

如果释放的是单回位，没有必要；释放的是群回位，在安排休息前看有没有回满的位置即可。

> 252信用经常不够买

满级宿舍和全 1 级宿舍每天只差 128 信用点，这个问题与 252 无关，另行讨论。

> 优化回满逻辑

应当区分“用于平衡心情的回满”和“暖机需要回满”。

> 通常使用条件是设置整组高优格雷伊回满。在此基础上如果有多个回满比如伊内丝稀音，刚好三者休息时间头接尾，则会导致长时间VIP位低于4人，导致其他需求4高优的套组无法换班。

自动化多设置一个低优先级就行，轮流吃群回位。问题在于 mower 现在休息到一半不能重新安排休息。

> 如果设置了第四人

为啥要设置第四人

**Reply by Shawnsdaddy** [reply]
得 24/7 确保VIP位置的单回被高效组吃到

这是不可能实现的，4 个单回位的恢复能力大于高效组的需求。

--  意思是说，，如果252一次只休息4个人的话，宿舍恢复能力是不够的
假设一次休息8个人，高效组休息会溢出的情况，可以根据当前高效组心情自动调节一次休息的人数
来确保24/7的VIP位都被高效组吃到

**Comment by ZhaoZuohong** [comment]
> 此基础上如果有多个回满比如伊内丝稀音

伊内丝可以设置低优先级，不占单回位

这也是自动低优先级不能实现的原因之一

**Comment by 383175186** [comment]
想到个新的优化点
但书 龙舌兰 的自动黑名单
心情＜4时剔除黑名单   心情＞20时加入黑名单
跑单的时候 就可以添加跑完后扫描心情

**Reply by Shawnsdaddy** [reply]
这个用附表可以解决
